plugins/woocommerce/client/legacy/css/twenty-nineteen.scss

```diff
@@ -33,19 +33,40 @@ a.button {
 }
 
 .woocommerce {
-	form .form-row {
-		.required {
-			color: firebrick;
-			text-decoration: none;
-			visibility: hidden; // Only show optional by default.
-
-			&[title] {
-				border: 0 !important;
-			}
+	form {
+		&.checkout_coupon .form-row:has(.input-with-wc-float-label) label {
+			transform: translateY(0.6em);
 		}
 
-		.optional {
-			visibility: visible;
+		.form-row {
+			.required {
+				color: firebrick;
+				text-decoration: none;
+				visibility: hidden; // Only show optional by default.
+	
+				&[title] {
+					border: 0 !important;
+				}
+			}
+	
+			.optional {
+				visibility: visible;
+			}
+	
+			&:has(.input-with-wc-float-label) {
+				position: relative;
+	
+				label {
+					@include woocommerce-float-label();
+					left: 0.8em;
+					transform: translateY(0.4em);
```

Comment:
Each Twenty* theme requires adjustments to better position the label inside the input.